trino/sqlalchemy/dialect.py

```diff
@@ -0,0 +1,301 @@
+        if not self.has_table(connection, table_name, schema):
+            raise exc.NoSuchTableError(f'schema={schema}, table={table_name}')
+        return self._get_columns(connection, table_name, schema, **kw)
+
+    def _get_columns(self, connection: Connection,
+                     table_name: str, schema: str = None, **kw) -> List[Dict[str, Any]]:
+        schema = schema or self._get_default_schema_name(connection)
+        query = dedent('''
+            SELECT
+                "column_name",
+                "column_default",
+                "is_nullable",
+                "data_type"
+            FROM "information_schema"."columns"
+            WHERE "table_schema" = :schema AND "table_name" = :table
+            ORDER BY "ordinal_position" ASC
+        ''').strip()
+        res = connection.execute(sql.text(query), schema=schema, table=table_name)
+        columns = []
+        for record in res:
+            column = dict(
+                name=record.column_name,
+                type=datatype.parse_sqltype(record.data_type),
+                nullable=(record.is_nullable or '').upper() == 'YES',
+                default=record.column_default,
+            )
+            columns.append(column)
+        return columns
+
+    def get_pk_constraint(self, connection: Connection,
+                          table_name: str, schema: str = None, **kw) -> Dict[str, Any]:
+        """Trino has no support for primary keys. Returns a dummy"""
+        return dict(name=None, constrained_columns=[])
+
+    def get_primary_keys(self, connection: Connection,
+                         table_name: str, schema: str = None, **kw) -> List[str]:
+        pk = self.get_pk_constraint(connection, table_name, schema)
+        return pk.get('constrained_columns')
+
+    def get_foreign_keys(self, connection: Connection,
+                         table_name: str, schema: str = None, **kw) -> List[Dict[str, Any]]:
+        """Trino has no support for foreign keys. Returns an empty list."""
+        return []
+
+    def get_schema_names(self, connection: Connection, **kw) -> List[str]:
+        query = 'SHOW SCHEMAS'
+        res = connection.execute(sql.text(query))
+        return [row.Schema for row in res]
+
+    def get_table_names(self, connection: Connection, schema: str = None, **kw) -> List[str]:
+        query = 'SHOW TABLES'
+        if schema:
+            query = f'{query} FROM {self.identifier_preparer.quote_identifier(schema)}'
+        res = connection.execute(sql.text(query))
+        return [row.Table for row in res]
+
+    def get_temp_table_names(self, connection: Connection, schema: str = None, **kw) -> List[str]:
+        """Trino has no support for temporary tables. Returns an empty list."""
+        return []
+
+    def get_view_names(self, connection: Connection, schema: str = None, **kw) -> List[str]:
+        schema = schema or self._get_default_schema_name(connection)
+        if schema is None:
```

Comment:
The sqlalchemy API docs say the schema is optional - we should treat it as optional in our method too.

Reply:
Optional means schema already provided in the `connection`.